src/renderer/components/data-settings/data-settings.js

```diff
@@ -228,6 +234,77 @@ export default Vue.extend({
       this.handleFreetubeImportFile(filePath)
     },
 
+    handleYoutubeCsvImportFile: function(filePath) { // first row = header, last row = empty
+      fs.readFile(filePath, async (err, data) => {
+        if (err) {
+          const message = this.$t('Settings.Data Settings.Unable to read file')
+          this.showToast({
+            message: `${message}: ${err}`
+          })
+          return
+        }
+        const textDecode = new TextDecoder('utf-8').decode(data)
+        console.log(textDecode)
+        const youtubeSubscriptions = textDecode.split('\n')
+        const primaryProfile = JSON.parse(JSON.stringify(this.profileList[0]))
+        const subscriptions = []
+
+        this.showToast({
+          message: this.$t('Settings.Data Settings.This might take a while, please wait')
+        })
+
+        this.updateShowProgressBar(true)
+        this.setProgressBarPercentage(0)
+        let count = 0
+        for (let i = 1; i < (youtubeSubscriptions.length - 1); i++) {
+          const channelId = youtubeSubscriptions[i].split(',')[0]
+          let channelInfo
+          if (this.backendPreference === 'invidious') { // only needed for thumbnail
+            channelInfo = await this.getChannelInfoInvidious(channelId)
+          } else {
+            channelInfo = await this.getChannelInfoLocal(channelId)
+          }
+
+          if (typeof channelInfo.author !== 'undefined') {
+            const subscription = {
+              id: channelId,
+              name: channelInfo.author,
+              thumbnail: channelInfo.authorThumbnails[1].url
+            }
+
+            const subExists = primaryProfile.subscriptions.findIndex((sub) => {
+              return sub.id === subscription.id || sub.name === subscription.name
```

Comment:
I copied this line from the other imports; is there a reason why there's a comparison of the subscription names? (Couldn't someone be subbed to different channels with the same name?). If that's the case (and it can be removed) then should "subExists" be checked before channelInfo is set to prevent unneeded requests(this goes for the other imports)?

Reply:
I added this to the PR, let me know if I should remove it